File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sIndex, tIndex = 0, 0
        while sIndex < len(s) and tIndex < len(t):
            if s[sIndex] == t[tIndex]:
                sIndex += 1
            
            tIndex += 1
            
        return sIndex == len(s)

File: 0392-is-subsequence/0392-is-subsequence.go
func isSubsequence(s string, t string) bool {
    i, j := 0, 0
    for j < len(t) && i < len(s) {
        if s[i] == t[j] {
            i += 1
        }

        j += 1
    }

    return i == len(s)
}